### PR TITLE
OLH-1622 Use the results from the API call to show a user their primary MFA method

### PR DESCRIPTION
**Pull request by saralk**
### What changed

- Create middleware to add the `mfaMethods` array to the session object. 
We take phone data from the user session object, and recreate the [logic from the template](https://github.com/govuk-one-login/di-account-management-frontend/blob/main/src/components/security/index.njk#L86-L94) to build the array. Once the work from [OLH-1621](https://govukverify.atlassian.net/jira/software/c/projects/OLH/boards/195?selectedIssue=OLH-1621) is complete, we will get the data from the API.
- Change the MFA section in the security page to render based on the contents of the `mfaMethods` object

### Why did it change

This is foundational work to enable the MFA management functionality 

### Environment variables or secrets
- [x] No environment variables or secrets were added or changed

- [ ] Application configuration is up-to-date
- [ ] Documented in the README
- [ ] Added to deployment steps
- [ ] Added to local startup config

## Testing

This is how it looked before, when the user had a verified phone

Screenshot 2024-03-26 at 14 55 41

And without a verified phone

Screenshot 2024-03-26 at 15 02 30

And on this PR, with a verified phone

Screenshot 2024-03-26 at 15 03 12

And with out a verified phone

Screenshot 2024-03-26 at 15 03 55

## How to review

[OLH-1621]: https://govukverify.atlassian.net/browse/OLH-1621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/security/security-controller.ts [hunk 1]
```diff
@@ -4,20 +4,61 @@ import { PATH_DATA } from "../../app.constants";
...
+      value: string,
+      actions = {};
+
+    if (method.mfaMethodType === "SMS") {
+      const phoneNumber = method.endPoint ? method.endPoint.slice(-4) : null;
+      key = req.t("pages.security.mfaSection.summaryList.phoneNumber.title");
+      value = req
+        .t("pages.security.mfaSection.summaryList.phoneNumber.value")
+        .replace("[phoneNumber]", phoneNumber);
+      actions = {
```
Comment (saralk) [inline src/components/security/security-controller.ts:26]: Once we have the functionality to manage the auth apps, we can refactor this a bit.

File: src/middleware/mfa-methods.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { NextFunction, Response, Request } from "express";
+
+export function mfaMethodsMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void {
+  if (req.session.user.isPhoneNumberVerified) {
```
Comment (andrew-moores) [inline src/middleware/mfa-methods.ts:8]: Do we need to do this? Won't the call to method-management tell us what the users PRIMARY method is explicitly without us needing to infer it like this?

Comment (saralk) [reply src/middleware/mfa-methods.ts:8]: yes, but right now, the call to the API doesn't exist, so this simulates it. 

If [Alfonso's PR](https://github.com/govuk-one-login/di-account-management-frontend/pull/1228) is merged in before, I will remove this middleware and use Alfonso's.

Comment (saralk) [reply src/middleware/mfa-methods.ts:8]: Once https://github.com/govuk-one-login/di-account-management-frontend/pull/1228/files is merged in, I can remove this file and use the new middleware.

File: src/components/security/tests/security-controller.test.ts
Comment (aenciso) [inline src/components/security/tests/security-controller.test.ts:1]: Perhaps add one more test for when mfaMethods is undefined and another one for when mfaMethodType is invalid?

**Review by andrew-moores — APPROVED** [review]

**Pull request merged**